### ISSUE TITLE
DuckDB exceeds memory_limit when ingesting `.csv.gz` file on `x64` Linux

### ISSUE THREAD

**Issue by k-anshul** [duckdb/duckdb#7888]
### What happens?

I am trying to ingest 33 `.csv.gz` files using `read_csv_auto`. Each file has compressed size ~120MB and uncompressed size ~350MB. The size of final db file is `2.5GB`.

I am setting a limit of `2GB` on duckdb before ingestion but the duckdb's memory usage jumps to 3.4GB and remains so even after ingestion (Interestingly it just frees 100MB on dropping table as well and remains at 3.3GB).

I tried with other combinations of `memory_limit` and input data as well and duckdb always exceeds the limit.
I see this issue only on a container running `linux` (debian v11 x64) and not on my laptop which is running macOS.

### To Reproduce

I can't share the data. Here are the commands that I am running. I check memory usage using top command. 

<img width="790" alt="image" src="https://github.com/duckdb/duckdb/assets/12948312/22a0960e-278e-4fb0-ae64-e85f41ab6396">

### OS:

debian (v11 x64)

### DuckDB Version:

v0.8.0 and latest master

### DuckDB Client:

CLI

### Full Name:

Anshul Khandelwal

### Affiliation:

Rill data

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by l1t1** [comment]
what does `describe test` return?

**Comment by k-anshul** [comment]
> what does `describe test` return?

```
┌─────────────────┬─────────────┬─────────┬─────────┬─────────┬───────┐
│   column_name   │ column_type │  null   │   key   │ default │ extra │
│     varchar     │   varchar   │ varchar │ varchar │ varchar │ int32 │
├─────────────────┼─────────────┼─────────┼─────────┼─────────┼───────┤
│ sxxxxxx         │ VARCHAR     │ YES     │         │         │       │
│ ixxxxxxxxxxxxxx │ VARCHAR     │ YES     │         │         │       │
│ sxxxxxxxxxx     │ BIGINT      │ YES     │         │         │       │
│ exxxxxxxxxx     │ BIGINT      │ YES     │         │         │       │
│ dxxxxxx         │ VARCHAR     │ YES     │         │         │       │
│ nxxxxxx         │ VARCHAR     │ YES     │         │         │       │
│ nxxxxxxxxx      │ BIGINT      │ YES     │         │         │       │
│ pxxxxxxxxxx     │ VARCHAR     │ YES     │         │         │       │
│ pxxxxxxxxxxxxx  │ BIGINT      │ YES     │         │         │       │
│ nxxxxxxxxx      │ VARCHAR     │ YES     │         │         │       │
│ nxxxxxxxxxxxx   │ BIGINT      │ YES     │         │         │       │
│ axxxxxxxxx      │ VARCHAR     │ YES     │         │         │       │
│ axxxxxxxxxxxx   │ BIGINT      │ YES     │         │         │       │
│ bxxxx           │ VARCHAR     │ YES     │         │         │       │
│ bxxxxxxx        │ BIGINT      │ YES     │         │         │       │
│ pxxxxxx         │ VARCHAR     │ YES     │         │         │       │
│ pxxxxxxxxx      │ BIGINT      │ YES     │         │         │       │
│ txxxxxxxxxxx    │ VARCHAR     │ YES     │         │         │       │
│ txxxxxxxxx      │ VARCHAR     │ YES     │         │         │       │
│ dxxxxxxx        │ BIGINT      │ YES     │         │         │       │
│ bxxxxxxx        │ VARCHAR     │ YES     │         │         │       │
│ cxxxxxxxxxxx    │ VARCHAR     │ YES     │         │         │       │
│ sxxxxxxxxxx     │ VARCHAR     │ YES     │         │         │       │
│ cxxx            │ VARCHAR     │ YES     │         │         │       │
│ pxxxxxxxxxx     │ VARCHAR     │ YES     │         │         │       │
│ fxxxxxx         │ VARCHAR     │ YES     │         │         │       │
├─────────────────┴─────────────┴─────────┴─────────┴─────────┴───────┤
│ 26 rows                                                   6 columns │
└─────────────────────────────────────────────────────────────────────┘

```

**Comment by szarnyasg** [comment]
@k-anshul do you have any updates on this issue? Is it still a problem with the latest `main` branch?

**Comment by k-anshul** [comment]
Yeah I can see the same issue on latest nightly branch as well. Interestingly now it is also reproducible on Mac-os where memory jumps beyond 2GB during ingestion and returns back to normal post ingestion. 
Adding a image that I generated using [memory-profiler](https://pypi.org/project/memory-profiler/) .
I also confirmed this behaviour with `top` command.
![image](https://github.com/duckdb/duckdb/assets/12948312/69f754f6-642d-4935-9c89-2a0ab34b83bd)
